setup/build_compile.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+
+# Build GMP
+function build_gmp() {
```

Comment:
It looks like there is an opportunity for some code reuse here with all these similar functions.